### PR TITLE
fix: fit addon proposing infinite dimensions

### PR DESCRIPTION
**Pull request by UziTech**
fixes https://github.com/xtermjs/xterm.js/issues/3085#issuecomment-690318136

### CODE REVIEW

File: addons/xterm-addon-fit/test/FitAddon.api.ts [hunk 1]
```diff
@@ -63,6 +63,11 @@ describe('FitAddon', () => {
         rows: 1
       });
     });
+
+    it('hidden', async function(): Promise<any> {
+      await loadFit(0, 0, false);
```
Comment (Tyriar) [inline addons/xterm-addon-fit/test/FitAddon.api.ts:68]: Should there also be a test for `loadFit(0, 0, true)`?

**Review by Tyriar — CHANGES_REQUESTED** [review]
Looks like the new test is failing in CI?

File: addons/xterm-addon-fit/test/FitAddon.api.ts (hunk 1 above)
Comment (UziTech) [reply addons/xterm-addon-fit/test/FitAddon.api.ts:68]: I'm not sure how to test this. Changing the terminal div to `display: none;` makes `fit()` fail in x-terminal but not in the tests. I'll have to dig deeper on x-terminal and see what is making the dimensions equal to 0.

Comment (UziTech) [reply addons/xterm-addon-fit/test/FitAddon.api.ts:68]: I found the issue. It seems that this issue only happens when the termina container div is hidden when it is attached to the terminal so the renderer doesn't have a chance to create a width before calling fit.

Now to figure out how to hide the terminal in the tests before attaching it to the terminal.

**Comment by UziTech** [comment]
It looks like this solves the issue of fit erroring when the terminal is attached to a hidden div but there are still other things that go wrong with the renderer. I will create another PR to address those issues.

File: addons/xterm-addon-fit/test/FitAddon.api.ts [hunk 2]
```diff
@@ -111,3 +123,8 @@ async function loadFit(width: number = 800, height: number = 450): Promise<void>
 async function unloadFit(): Promise<void> {
   await page.evaluate(`window.fit.dispose();`);
 }
+
+async function resetTerminal(): Promise<void> {
+  await page.evaluate(`window.term.dispose()`);
+  await page.evaluate(`window.term.open(document.querySelector('#terminal-container'))`);
+}
```
Comment (Tyriar) [inline addons/xterm-addon-fit/test/FitAddon.api.ts:130]: We can instead move the `openTerminal` call into a `beforeEach` so the standard setup happens for each test, it's a pretty cheap action.

File: addons/xterm-addon-fit/test/FitAddon.api.ts [hunk 3]
```diff
@@ -63,6 +63,18 @@ describe('FitAddon', () => {
...
     });
+
+    it('hidden', async function(): Promise<any> {
+      try {
+        await page.evaluate(`document.querySelector('#terminal-container').style.display='none'`);
+        await resetTerminal();
+        await loadFit();
+        assert.equal(await page.evaluate(`window.fit.proposeDimensions()`), undefined);
+      } finally {
+        await page.evaluate(`document.querySelector('#terminal-container').style.display=''`);
```
Comment (Tyriar) [inline addons/xterm-addon-fit/test/FitAddon.api.ts:74]: This display change could be moved into the `beforeEach` instead, that's always seemed safer than relying on try/finally to me to handle resetting things.

**Comment by UziTech** [comment]
I created #3089 to handle the other issues created when the terminal is attached to a hidden div.

**Review by Tyriar — APPROVED** [review]

**Pull request merged**